recipes/ccfits/all/conanfile.py

```diff
@@ -21,59 +26,56 @@ class CcfitsConan(ConanFile):
         "fPIC": True,
     }
 
-    generators = "cmake"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("cfitsio/4.0.0")
+        self.requires("cfitsio/4.1.0")
 
-    def validate(self):
-        if tools.Version(self.version) >= "2.6":
+    def validate_build(self):
```

Comment:
Does no this need to go everywhere? https://docs.conan.io/en/latest/reference/conanfile/methods.html#validate-build

Thats how we were using the old `valdiate()`

Reply:
@SpaceIm there is an open issue about it if you want to comment: https://github.com/conan-io/conan/issues/12210

Also, there is a PR to update the documentation: https://github.com/conan-io/docs/pull/2778

Reply:
I've moved this check to `validate_build()` because C++11 is required at build time, not consume time, that's all.

Reply:
That's make sense. Thanks!